lib/feed/bloc/feed_bloc.dart

```diff
@@ -283,7 +283,33 @@ class FeedBloc extends Bloc<FeedEvent, FeedState> {
           return emit(state.copyWith(status: FeedStatus.failure));
         }
       case PostAction.report:
-      // TODO: Handle this case.
+        // Optimistically report the post
+        int existingPostViewMediaIndex = state.postViewMedias.indexWhere((PostViewMedia postViewMedia) => postViewMedia.postView.post.id == event.postId);
+
+        PostViewMedia postViewMedia = state.postViewMedias[existingPostViewMediaIndex];
+        PostView originalPostView = postViewMedia.postView;
+
+        try {
+          PostView updatedPostView = optimisticallyReportPost(postViewMedia, true);
+          state.postViewMedias[existingPostViewMediaIndex].postView = updatedPostView;
+
+          emit(state.copyWith(status: FeedStatus.fetching));
+
+          bool success = await reportPost(originalPostView.post.id, event.value);
+
+          //mark post as read
+          if (success) add(FeedItemActionedEvent(postAction: PostAction.read, postId: originalPostView.post.id, value: true));
+
+          // Restore the original post contents if not successful
```

Comment:
We should return early here if it's successful, or wrap the remaining lines in an else condition. Otherwise, we might trigger `FeedStatus.failure` when the action was performed successfully.